migration/bucket_migrator_test.go

```diff
@@ -0,0 +1,84 @@
+package migration_test
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/db/pebble"
+	"github.com/NethermindEth/juno/migration"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBucketMover(t *testing.T) {
+	beforeCalled := false
+	sourceBucket := db.Bucket(0)
+	destBucket := db.Bucket(1)
+	mover := migration.NewBucketMover(sourceBucket, destBucket).WithBefore(func() {
+		beforeCalled = true
+	}).WithBatchSize(2).WithKeyFilter(func(b []byte) (bool, error) {
+		return len(b) > 1, nil
+	})
+
+	testDB := pebble.NewMemTest()
+	require.NoError(t, testDB.Update(func(txn db.Transaction) error {
+		for i := byte(0); i < 10; i++ {
+			if err := txn.Set(sourceBucket.Key([]byte{i}), []byte{i}); err != nil {
+				return err
+			}
+		}
+		return txn.Set(sourceBucket.Key(), []byte{44})
+	}))
+
+	mover.Before()
+	require.True(t, beforeCalled)
+
+	txnBudget := 100
+	for {
```

Comment:
nit: is it possible to remove this loop from the test? Maybe decreasing the number of elements in the db from `10` to `4` would make that reasonable?

Reply:
Done